feather/common/src/world.rs

```diff
@@ -149,6 +153,248 @@ impl World {
         self.chunk_map.block_at(pos)
     }
 
+    pub fn adjacent_block_cubic(&self, pos: BlockPosition, dir: FacingCubic) -> Option<BlockId> {
+        self.block_at(
+            pos.adjacent(match dir {
+                FacingCubic::North => BlockFace::North,
+                FacingCubic::East => BlockFace::East,
+                FacingCubic::South => BlockFace::South,
+                FacingCubic::West => BlockFace::West,
+                FacingCubic::Up => BlockFace::Top,
+                FacingCubic::Down => BlockFace::Bottom,
+            })
+            .try_into()
+            .unwrap(),
+        )
+    }
+
+    pub fn adjacent_block_cardinal(
+        &self,
+        pos: BlockPosition,
+        dir: FacingCardinal,
+    ) -> Option<BlockId> {
+        self.adjacent_block_cubic(pos, dir.to_facing_cubic())
+    }
+
+    pub fn adjacent_block_cardinal_and_down(
+        &self,
+        pos: BlockPosition,
+        dir: FacingCardinalAndDown,
+    ) -> Option<BlockId> {
+        self.adjacent_block_cubic(pos, dir.to_facing_cubic())
+    }
+
+    pub fn get_facing_block(&self, pos: BlockPosition) -> Option<BlockId> {
+        let block = self.block_at(pos.try_into().unwrap())?;
+        let a = block.facing_cardinal().map(FacingCardinal::to_facing_cubic);
+        let b = block
+            .facing_cardinal_and_down()
+            .map(FacingCardinalAndDown::to_facing_cubic);
+        let c = block.facing_cubic();
+        let dir = [a, b, c].iter().find_map(|&e| e)?;
```

Comment:
There's `BlockId::has_facing_cardinal()` and other `has_*` functions, so you can do the same thing in an if-else style, and it'll become much more readable.

Reply:
The if else chain version is a huge block of text, but I agree that it is easier to understand.